iroha/src/smartcontracts/isi/asset.rs

```diff
@@ -159,6 +184,30 @@ pub mod isi {
         }
     }
 
+    impl<W: WorldTrait> Execute<W> for Burn<Asset, Fixed> {
+        type Error = Error;
+
+        fn execute(
+            self,
+            _authority: <Account as Identifiable>::Id,
+            wsv: &WorldStateView<W>,
+        ) -> Result<(), Error> {
+            assert_asset_type(
+                &self.destination_id.definition_id,
+                wsv,
+                AssetValueType::Fixed,
+            )?;
+            wsv.modify_asset(&self.destination_id, |asset| {
+                let quantity: &mut Fixed = asset.try_as_mut()?;
+                *quantity = quantity
+                    .checked_sub(self.object)
+                    .ok_or_else(|| error!("Not enough quantity to burn."))?;
```

Comment:
Could you add this error to math error?
https://github.com/hyperledger/iroha/blob/00ba1876f1f102af141c7f92bcd33273ad601c09/iroha/src/smartcontracts/isi/mod.rs#L105-L109

Reply:
Done.